src/components/DatePicker/index.android.js

```diff
@@ -8,87 +8,76 @@ import CONST from '../../CONST';
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
 
     /**
      * @param {Event} event
      * @param {Date} selectedDate
      */
-    setDate(event, selectedDate) {
-        this.setState({isPickerVisible: false});
+    function setDate(event, selectedDate) {
+        setIsPickerVisible(false);
 
         if (event.type === 'set') {
             const asMoment = moment(selectedDate, true);
-            this.props.onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
+            onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
         }
     }
 
-    showPicker() {
+    function showPicker() {
         Keyboard.dismiss();
-        this.setState({isPickerVisible: true});
+        setIsPickerVisible(true);
     }
 
-    render() {
-        const dateAsText = this.props.value || this.props.defaultValue ? moment(this.props.value || this.props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
+    const dateAsText = value || defaultValue ? moment(value || defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
 
-        return (
-            <>
-                <TextInput
-                    label={this.props.label}
-                    accessibilityLabel={this.props.label}
-                    accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
-                    value={dateAsText}
-                    forceActiveLabel
-                    placeholder={this.props.placeholder}
-                    errorText={this.props.errorText}
-                    containerStyles={this.props.containerStyles}
-                    textInputContainerStyles={this.state.isPickerVisible ? [styles.borderColorFocus] : []}
-                    onPress={this.showPicker}
-                    editable={false}
-                    disabled={this.props.disabled}
-                    onBlur={this.props.onBlur}
-                    ref={(el) => {
-                        if (!_.isFunction(this.props.innerRef)) {
-                            return;
-                        }
-                        if (el && el.focus && typeof el.focus === 'function') {
-                            let inputRef = {...el};
-                            inputRef = {...inputRef, focus: this.showPicker};
-                            this.props.innerRef(inputRef);
-                            return;
-                        }
+    return (
+        <>
+            <TextInput
+                label={label}
+                accessibilityLabel={label}
+                accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
+                value={dateAsText}
+                forceActiveLabel
+                placeholder={placeholder}
+                errorText={errorText}
+                containerStyles={containerStyles}
+                textInputContainerStyles={isPickerVisible ? [styles.borderColorFocus] : []}
+                onPress={() => showPicker()}
```

Comment:
```suggestion
                onPress={showPicker}
```

It could be simplified.

Reply:
Replied [here](https://github.com/Expensify/App/pull/25657#discussion_r1301672343)